### PR TITLE
Swatch: Convert component to TypeScript

### PR DESCRIPTION
**Pull request by walbo**
## What?
Refactor the `Swatch` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactored the component TypeScript

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Test the various use of Swatch within the block/site editors in Gutenberg still work

### CODE REVIEW

File: packages/components/src/swatch/index.tsx [hunk 1]
```diff
@@ -7,8 +7,9 @@ import { swatch } from '@wordpress/icons';
  * Internal dependencies
  */
 import Icon from '../icon';
+import type { SwatchProps } from './types';
 
-function Swatch( { fill } ) {
+function Swatch( { fill }: SwatchProps ) {
```
Comment (mirka) [inline packages/components/src/swatch/index.tsx:14]: Oof, another surprise component I didn't know about 😂 I think we'll need to consolidate this with `ColorIndicator`. Good thing `Swatch` isn't exported in the public package though.

Could we add a warning comment like this?

```suggestion
// This component will be deprecated. Use `ColorIndicator` instead.
// TODO: Consolidate this component with `ColorIndicator`.
function Swatch( { fill }: SwatchProps ) {
```

cc @ciampo

**Review by mirka — APPROVED** [review]
Looks good 👍

File: packages/components/src/swatch/index.tsx (hunk 1 above)
Comment (ciampo) [reply packages/components/src/swatch/index.tsx:14]: Didn't know about this component either 😅 

Some time ago I had drafted a [plan of action](https://github.com/WordPress/gutenberg/issues/33820) to tidy up color-related components, but we never had the resources to actually work on that plan

Comment (walbo) [reply packages/components/src/swatch/index.tsx:14]: Yep. I think this component can be deprecated. Looks like its only are used in a couple of places (`DuotonePicker` and `ColorListPicker`)

Comment (ciampo) [reply packages/components/src/swatch/index.tsx:14]: Yup — we can definitely deprecate it and even re-implement it by using `ColorIndicator` under the hood.

**Comment by ciampo** [comment]
Hey @walbo , after merging this PR, do you think you would you be able to work on aliasing this component to `ColorIndicator` and marking it as deprecated as agreed in https://github.com/WordPress/gutenberg/pull/42162#discussion_r920522965 ? Thank you in advance and no worries in case you don't have capacity 🙏

**Comment by walbo** [comment]
> Hey @walbo , after merging this PR, do you think you would you be able to work on aliasing this component to `ColorIndicator` and marking it as deprecated as agreed in [#42162 (comment)](https://github.com/WordPress/gutenberg/pull/42162#discussion_r920522965) ? Thank you in advance and no worries in case you don't have capacity 🙏

Yes, I can take a look at it next week.

I'll resolve the merge conflicts and get this merged today.

**Pull request merged**